src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Extensions/SystemPropertiesCollectionExtensions.cs

```diff
@@ -11,13 +10,14 @@ static internal class SystemPropertiesCollectionExtensions
     {
         internal static IDictionary<string, object> ToDictionary(this SystemPropertiesCollection collection)
         {
-            return new Dictionary<string, object>()
-            {
-                { "SequenceNumber", collection.SequenceNumber},
-                { "Offset", collection.Offset },
-                { "PartitionKey", collection.PartitionKey },
-                { "EnqueuedTimeUtc", collection.EnqueuedTimeUtc }
-            };
+            IDictionary<string, object> modifiedDictionary = collection;
+            
+            // Following is needed to maintain structure of bindingdata: https://github.com/Azure/azure-webjobs-sdk/pull/1849
+            modifiedDictionary["SequenceNumber"] = collection.SequenceNumber;
+            modifiedDictionary["Offset"] = collection.Offset;
+            modifiedDictionary["PartitionKey"] = collection.PartitionKey;
+            modifiedDictionary["EnqueuedTimeUtc"] = collection.EnqueuedTimeUtc;
+            return modifiedDictionary;
         }
```

Comment:
Is it right to be modifying the SystemPropertiesCollection that was passed in here? Feels like we should be making a copy of that collection, and then modifying the copy. Let me know if I'm missing something.

Reply:
You are right. I did not intend to change the original collection. Will send follow up PR.

Reply:
Preferable, I agree. That way we aren't modifying the instance we got from EventData.SystemProperties.